marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -151,11 +151,22 @@ protected JsonNode determineIngestionEndpointParams(Map<String, String> params)
         }
 
         if (!endpointParams.hasNonNull("workUnit")) {
-            ObjectNode defaultWorkUnit = objectMapper.createObjectNode();
-            defaultWorkUnit.put("uriprefix", params.get("uriprefix")!=null ? params.get("uriprefix"):"");
+            defaultWorkUnit = objectMapper.createObjectNode();
+            if (params.get("uriprefix") == null) {
```

Comment:
The endpoint should do this, not the connector. The endpoint should say - if uriprefix is in the workUnit, then I'll use that, even if it's null. It's up to the client to pass in the correct value for uriprefix - if the client passes in "uriprefix": null, then the endpoint says - Well I guess you want null as the prefix, so I'll use that.

Reply:
Made the changes